enclave/core/sgx/entropy.c

```diff
@@ -1,48 +1,82 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-#include <openenclave/bits/safecrt.h>
+#include <openenclave/corelibc/string.h>
 #include <openenclave/enclave.h>
 #include <openenclave/internal/entropy.h>
+#include <openenclave/internal/raise.h>
 #include <openenclave/internal/rdrand.h>
+#include <openenclave/internal/rdseed.h>
+#include "cpuid.h"
 
-/* TODO: This should use RDSEED instead. See issue #242. */
-oe_result_t oe_get_entropy(void* output, size_t len)
+typedef uint64_t (*_entropy_function_t)(void);
+
+OE_INLINE bool _has_cpuid_feature(
```

Comment:
Does not seem like inline is needed here

Reply:
It's not strictly necessary, but it's a reasonable compiler hint since this method is broken out for readability and could be optimized by its inclusion into `oe_entropy_kind_t`. It also includes the `static` keyword which is convenient here.